test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -299,10 +299,14 @@ static void ssl_test_check_handshaker_peers(tsi_test_fixture* fixture) {
   GPR_ASSERT(ssl_fixture->key_cert_lib != nullptr);
   ssl_key_cert_lib* key_cert_lib = ssl_fixture->key_cert_lib;
   tsi_peer peer;
-  bool expect_success =
+  bool expect_server_success =
       !(key_cert_lib->use_bad_server_cert ||
         (key_cert_lib->use_bad_client_cert && ssl_fixture->force_client_auth));
-  if (expect_success) {
+  // In TLS 1.3, client side handshake succeeds even if the client sends a bad
```

Comment:
If that's the case, can we only do a special handling for TLS.1.3 and keep using the original check for TLS 1.2 and below?

Reply:
Ah I just noticed your comment (i.e., there is no API exposing the negotiated TLS version), in which case, I agree we should not perform any version-specific check here. Please ignore my comment above.

Reply:
I'm probably missing some context but I think |SSL_version| can be used to get the negotiated version of the handshake is complete:
https://www.openssl.org/docs/man1.1.1/man3/SSL_version.html

Reply:
Yeah, I think the issue is we do not expose SSL_CTX in `ssl_transport_security.h` so we are not able to call `SSL_version()` API in the unit test. Alternatively, we can create a new API that wraps `SSL_version()` but I am not sure it is worthwhile just for the test.

Reply:
Beside Yihua's point that SSL_CTX is not exposed, SSL_get_version() is only available on OpenSSL 1.1 or above. gRPC needs to support OpenSSL 1.0.2.

We will need to think about how to address this test properly.

Reply:
Ah yes that makes sense. I didn't notice that this information is needed in the test and not the code itself.